src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -30,10 +31,11 @@
 import java.util.Optional;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import javax.annotation.Nullable;
 import jenkins.model.Jenkins;
 
 public class ComputeEngineInstance extends AbstractCloudSlave {
-  private static final long serialVersionUID = 1;
+  private static final long serialVersionUID = 2L;
```

Comment:
With that change all old configs will work?
I must say Im not sure of that..

If we want to keep it we should create test for migrating..

Reply:
The AbstractCloudSlave instances are meant to be ephemeral, therefore they shouldn't be getting seriliazed to the config.xml unless I'm misunderstanding something.

Reply:
They are not ephemeral..

Try adding few slaves and restarting instance.. :)

Reply:
Ah ok, good to know. In that case I've leave the UID unchanged. Older versions of the serialized object will be instantiated with a null value which we account for in the getJavaExecPathOrDefault() method.